src/extension/src/azure/azure-app-service/appServiceProvider.ts

```diff
@@ -0,0 +1,78 @@
+import { SubscriptionItem } from "../azure-auth/azureAuth";
+import { WebSiteManagementClient } from "azure-arm-website";
+import { ServiceClientCredentials } from "ms-rest";
+import { SubscriptionError } from "../../errors";
+import { CONSTANTS, AppType } from "../../constants";
+import { AppNameValidationResult, NameValidator } from "../utils/nameValidator";
+
+export class AppServiceProvider {
+  private webClient: WebSiteManagementClient | undefined;
+
+  /*
+   * Sets a web client from a users selected subscription item's credentials
+   */
+  private setWebClient(userSubscriptionItem: SubscriptionItem): void {
+    if (
+      this.webClient === undefined ||
+      this.webClient.subscriptionId !== userSubscriptionItem.subscriptionId
+    ) {
+      this.webClient = this.createWebClient(userSubscriptionItem);
+    }
+  }
+
+  private createWebClient(
+    userSubscriptionItem: SubscriptionItem
+  ): WebSiteManagementClient {
+    const credentials: ServiceClientCredentials =
+      userSubscriptionItem.session.credentials;
+
+    if (
+      userSubscriptionItem === undefined ||
+      userSubscriptionItem.subscription === undefined ||
+      userSubscriptionItem.subscriptionId === undefined
+    ) {
+      throw new SubscriptionError(CONSTANTS.ERRORS.SUBSCRIPTION_NOT_DEFINED);
+    }
+
+    return new WebSiteManagementClient(
+      credentials,
+      userSubscriptionItem.subscriptionId
+    );
+  }
+
+  public async checkWebAppName(
+    appName: string,
+    subscriptionItem: SubscriptionItem
+  ): Promise<string | undefined> {
+    try {
+      this.setWebClient(subscriptionItem);
```

Comment:
Just a minor personal thought, it feels like createWebClient and setWebClient could be merged into 1 function. Since you throw error in createWebClient and handle the error when you call setWebClient. Also, in your setWebClient, you did a check and if true, you createWebClient.

Reply:
We will need to refactor some of the code in another PR. 
This is currently what we have for Functions and Cosmos DB too, so I don't want to change the logics in this PR.